### PR TITLE
TTOOLS-598 SQL editor highlighting and autocomplete

### PR DESCRIPTION
**Pull request by mdrillin**
Would just like comments on these changes.  
The intention is to enable SQL highlighting for the codemirror component, and enable auto-complete hints.  The hightlighting is working ok, but auto-complete hints seem to be hidden, at least in dev environment.

### CODE REVIEW

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/2441/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

File: app/ui-react/packages/ui/src/Data/Virtualizations/Views/ViewEditContent.tsx [hunk 1]
```diff
@@ -1,9 +1,21 @@
+import 'codemirror/lib/codemirror.css';
```
Comment (gashcrumb) [inline app/ui-react/packages/ui/src/Data/Virtualizations/Views/ViewEditContent.tsx]: Yup, as @riccardo-forina said we should move all these code-mirror specific imports/requires into the `TextEditor` component.

File: app/ui-react/packages/ui/src/Data/Virtualizations/Views/ViewEditContent.css [hunk 2]
```diff
@@ -14,3 +14,8 @@
   margin-left: auto;
   width: 90%;
 }
+
+.CodeMirror-hints {
```
Comment (gashcrumb) [inline app/ui-react/packages/ui/src/Data/Virtualizations/Views/ViewEditContent.css]: Should put this in the `TextEditor` component's CSS too, probably be good to increase the specificity just to be on the safe side.

File: app/ui-react/packages/ui/src/Data/Virtualizations/Views/ViewEditContent.tsx (hunk 1 above)
Comment (mdrillin) [reply app/ui-react/packages/ui/src/Data/Virtualizations/Views/ViewEditContent.tsx]: moved into TextEditor and amended the PR

File: app/ui-react/packages/ui/src/Data/Virtualizations/Views/ViewEditContent.css (hunk 2 above)
Comment (mdrillin) [reply app/ui-react/packages/ui/src/Data/Virtualizations/Views/ViewEditContent.css]: moved and amended the PR

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/2451/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Comment by gashcrumb** [comment]
Ah, this will need a rebase, @riccardo-forina did some much needed refactoring of the text editor component to sort out some problems.

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/2496/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Comment by rareddy** [comment]
@riccardo-forina I think still the following import is missing from refactoring.

`import 'codemirror/addon/hint/show-hint.css';`

**Comment by riccardo-forina** [comment]
Sure, just open a new PR with that add!

**Pull request closed without merging**